src/test/java/tests/web/InventorySortingTest.java

```diff
@@ -1,90 +1,45 @@
-            }
-        }
-        return true;
-    }
-
-    private boolean isSortedDescending(List<? extends Comparable> list) {
-        for (int i = 0; i < list.size() - 1; i++) {
-            if (list.get(i).compareTo(list.get(i + 1)) < 0) {
-                return false;
-            }
-        }
-        return true;
-    }
-
-    @Test
-    public void testInventorySortingByNameAZ() {
-        LOGGER.info("Starting test: testInventorySorting");
-        login(driver, "standard_user", "secret_sauce");
-
-        InventoryPage inventoryPage = new InventoryPage(driver);
-
-        inventoryPage.selectSortingOption("Name (A to Z)");
-        List<String> sortedNamesAZ = inventoryPage.getProductNames();
-        LOGGER.info("Sorted product names (A to Z): {}", sortedNamesAZ);
-        Assert.assertTrue(isSortedAscending(sortedNamesAZ), "Products are sorted correctly by Name (A to Z)");
-    }
-
-    @Test
-    public void testInventorySortingByNameZA() {
-        LOGGER.info("Starting test: testInventorySorting");
-        login(driver, "standard_user", "secret_sauce");
-
-        InventoryPage inventoryPage = new InventoryPage(driver);
-
-        inventoryPage.selectSortingOption("Name (Z to A)");
-        List<String> sortedNamesZA = inventoryPage.getProductNames();
-        LOGGER.info("Sorted product names (Z to A): {}", sortedNamesZA);
-        Assert.assertTrue(isSortedDescending(sortedNamesZA), "Products are sorted correctly by Name (Z to A)");
-    }
-
-    @Test
-    public void testInventorySortingByPriceLowHigh() {
-        LOGGER.info("Starting test: testInventorySorting");
-        login(driver, "standard_user", "secret_sauce");
-
-        InventoryPage inventoryPage = new InventoryPage(driver);
-
-        inventoryPage.selectSortingOption("Price (low to high)");
-        List<Double> sortedPricesLowToHigh = inventoryPage.getProductPrices();
-        LOGGER.info("Sorted product prices (Low to High): {}", sortedPricesLowToHigh);
-        Assert.assertTrue(isSortedAscending(sortedPricesLowToHigh), "Products are sorted correctly by Price (Low to High)");
+public class InventorySortingTest extends BaseTest {
+    @DataProvider(name = "sortingOptions")
+    public Object[][] sortingOptions() {
+        return new Object[][] {
+                { "Name (A to Z)", true, false },
+                { "Name (Z to A)", false, true },
+                { "Price (low to high)", true, false },
+                { "Price (high to low)", false, true }
+        };
     }
 
-    @Test
-    public void testInventorySortingByPriceHighLow(){
-        LOGGER.info("Starting test: testInventorySorting");
-        login(driver, "standard_user", "secret_sauce");
-
-        InventoryPage inventoryPage = new InventoryPage(driver);
+    @Test(dataProvider = "sortingOptions")
+    public void testInventorySortingByNameAZ(String sortingOption, boolean isAscending, boolean isDescending) {
+        InventoryPage inventoryPage = new InventoryPage(getDriver());
```

Comment:
Here I left a good example of how test methods look like
 public void verifyProductSortInAscendingOrder() {
        ProductListBasePage productPage = loginToAccount();
        productPage.sortProduct(SortType.PRICE_LOW_TO_HIGH);
        Assert.assertTrue(productPage.verifyProductSortingByPrice(SortType.PRICE_LOW_TO_HIGH), "Products are not sorted correctly");
    }

    @Test
    public void verifyProductSortInDescendingOrder() {
        ProductListBasePage productPage = loginToAccount();
        productPage.sortProduct(SortType.PRICE_HIGH_TO_LOW);
        Assert.assertTrue(productPage.verifyProductSortingByPrice(SortType.PRICE_HIGH_TO_LOW), "Products are not sorted correctly");
    }